goSrc/src/vmCaller/iroha/commands.go

```diff
@@ -97,6 +97,248 @@ func AddAssetQuantity(asset string, amount string) error {
+	if err != nil {
+		return err
+	}
+	if commandResult.error_code != 0 {
+		error_extra := ""
+		error_extra_ptr := commandResult.error_extra.toStringAndRelease()
+		if error_extra_ptr != nil {
+			error_extra = ": " + *error_extra_ptr
+		}
+		return fmt.Errorf("Error executing SetAccountQuorum command: %s", error_extra)
+	}
+	return nil
+}
+
+func AddSignatory(account string, key string) error {
+	command := &pb.Command{Command: &pb.Command_AddSignatory{
+		AddSignatory: &pb.AddSignatory{
+			AccountId: account,
+			PublicKey: key,
+		}}}
+	commandResult, err := makeProtobufCmdAndExecute(IrohaCommandExecutor, command)
+	if err != nil {
+		return err
+	}
+	if commandResult.error_code != 0 {
+		error_extra := ""
+		error_extra_ptr := commandResult.error_extra.toStringAndRelease()
+		if error_extra_ptr != nil {
+			error_extra = ": " + *error_extra_ptr
+		}
+		return fmt.Errorf("Error executing AddSignatory command: %s", error_extra)
+	}
+
+	return nil
+}
+
+func RemoveSignatory(account string, key string) error {
+	command := &pb.Command{Command: &pb.Command_RemoveSignatory{
+		RemoveSignatory: &pb.RemoveSignatory{
+			AccountId: account,
+			PublicKey: key,
+		}}}
+	commandResult, err := makeProtobufCmdAndExecute(IrohaCommandExecutor, command)
+	if err != nil {
+		return err
+	}
+	if commandResult.error_code != 0 {
+		error_extra := ""
+		error_extra_ptr := commandResult.error_extra.toStringAndRelease()
+		if error_extra_ptr != nil {
+			error_extra = ": " + *error_extra_ptr
+		}
+		return fmt.Errorf("Error executing RemoveSignatory command: %s", error_extra)
+	}
+
+	return nil
+}
+
+func CreateDomain(domain string, role string) error {
+	command := &pb.Command{Command: &pb.Command_CreateDomain{
+		CreateDomain: &pb.CreateDomain{
+			DomainId:    domain,
+			DefaultRole: role,
+		}}}
+	commandResult, err := makeProtobufCmdAndExecute(IrohaCommandExecutor, command)
+	if err != nil {
+		return err
+	}
+	if commandResult.error_code != 0 {
+		error_extra := ""
+		error_extra_ptr := commandResult.error_extra.toStringAndRelease()
+		if error_extra_ptr != nil {
+			error_extra = ": " + *error_extra_ptr
+		}
+		return fmt.Errorf("Error executing CreateDomain command: %s", error_extra)
+	}
+
+	return nil
+}
+
+func CreateAsset(name string, domain string, precision string) error {
+	precision_uint, err := strconv.ParseUint(precision, 10, 32)
+	command := &pb.Command{Command: &pb.Command_CreateAsset{
+		CreateAsset: &pb.CreateAsset{
+			AssetName: name,
+			DomainId:  domain,
+			Precision: uint32(precision_uint),
+		}}}
+	commandResult, err := makeProtobufCmdAndExecute(IrohaCommandExecutor, command)
+	if err != nil {
+		return err
+	}
+	if commandResult.error_code != 0 {
+		error_extra := ""
+		error_extra_ptr := commandResult.error_extra.toStringAndRelease()
+		if error_extra_ptr != nil {
+			error_extra = ": " + *error_extra_ptr
+		}
+		return fmt.Errorf("Error executing CreateAsset command: %s", error_extra)
+	}
+
+	return nil
```

Comment:
All command differs the only command structure and command name in log output. It seems can be refactored to a single implementation of this logic in some function. And calls of the  function elsewhere.

Also queries can be refactored the same way.

Reply:
@iceseer  Are you suggesting to have an error handling part(i.e. line 285 to 295 )in  a function?

Reply:
> @iceseer Are you suggesting to have an error handling part(i.e. line 285 to 295 )in a function?

@Ayush-Jalan I think he suggested to extract function for all commands, because they are quite similar:
![image](https://user-images.githubusercontent.com/15332594/130181178-a84dcdc2-854d-4274-94cd-91a9d85b8ab6.png)
And call the function from all integrated commands.

Reply:
I have refactored the commands. Please, have a look 
@iceseer  @baziorek

Reply:
Refactoring for queries was not possible since they differ largely for each Query